components/layout_2020/flow/inline/mod.rs

```diff
@@ -2344,8 +2348,21 @@ impl<'a> ContentSizesComputation<'a> {
                         self.had_content_yet = true;
                     }
                 }
+
+                self.prevent_soft_wrap_opportunity_before_next_atomic =
+                    text_run.prevent_soft_wrap_opportunity_at_end;
             },
             InlineItem::Atomic(atomic) => {
+                let soft_wrap_opportunity_prevented = mem::replace(
+                    &mut self.prevent_soft_wrap_opportunity_before_next_atomic,
+                    false,
+                );
+
+                // TODO: need to handle TextWrapMode::Nowrap.
+                if self.previous_item_was_atomic_inline && soft_wrap_opportunity_prevented {
```

Comment:
Is the conditional here correct? Should `soft_wrap_opportunity_prevented` be negated? Currently the logic will call `line_break_opportunity` when it is `true`. For the 2nd atomic item,` previous_item_was_atomic_inline` will be `true` but `soft_wrap_opportunity_prevented` is `false` since the 1st atomic item would have reset the flag to `false`.